jump/src/lib.rs

```diff
@@ -217,16 +222,24 @@ pub fn prepare_boot(scie_jump_version: &str) -> Result<BootAction, String> {
     }
     let payload = &data[jump.size..data.len() - lift.size];
     let installer = Installer::new(payload);
-    match context::select_command(&current_exe, &jump, &lift, &installer) {
+    match context::select_command(&current_exe, &jump, &lift, &installer, custom_env) {
         Ok(selected_command) => {
             installer.install(&selected_command.files)?;
             let process = selected_command.process;
-            trace!("Prepared {process:#?}");
-            env::set_var("SCIE", current_exe.exe.as_os_str());
-            env::set_var("SCIE_ARGV0", current_exe.invoked_as.as_os_str());
+            let extra_env: Vec<(OsString, Option<OsString>)> = vec![
+                ("SCIE".into(), Some(current_exe.exe.into_os_string())),
+                (
+                    "SCIE_ARGV0".into(),
+                    Some(current_exe.invoked_as.into_os_string()),
+                ),
+                // Avoid subprocesses that re-execute this SCIE unintentionally getting in an
+                // infinite loop.
+                ("SCIE_BOOT".into(), None),
+            ];
```

Comment:
This got SCIE_BOOT nuking from here: https://github.com/a-scie/jump/pull/293/files#r2008480914

The non-locality of SCIE_BOOT considerations is a loss, but the non-mutation of env vars is a more important safety win / sanity-preservation win.